### DISCUSSION TITLE
Problems with long bytestreams

### DISCUSSION THREAD

**Discussion by Rustimation** [philippoo66/optolink-splitter#45] [Q&A] [answered]
Hi Phil,
Using Parameter Address Table VScotHO1_200_11 Excel as produced by Frans yesterday.
-->Reading 212 Bytes as a string  from address 0x9020 to get weekly consumption heating in kW (53 weeks). 
  Python settings_ini.py: ("HeizgVerbrWoche", 0x9020, 212, 'raw'), 
  or in MQTT Explorer: read;0c9020;212;raw
Lateron to be converted to decimal (litttle endian).
--> Expected behaviour: 212 bytes (424) Nibbles or halfBytes) are read and shown in MQTT Explorer and processed elsewhere.
--> Observed behaviour: 55 bytes (110 nibbles) are shown -- Any byte string > 55 bytes is cut down to 55 bytes.

I checked MQTT Explorer - it can handle 65536 bytes per string, so 212 bytes shouldn't be a problem
I tried an older paho-mqtt client on another machine, works as expected
I tried a python test script using the paho-mqtt client installed on the optolink server, works as expected. 212 bytes published and received.

Now I'm stuck somehow. What could be the problem? Do some of the inner workings of the Optolink Splitter cut down long strings?
Since I'm not really fluent in Python, I didn't analyze your code any further.

As a workaround I'm reading the parameter in 4 chunks - nor really efficient.

**Comment by Rustimation** [comment]
I used the following Testscript
```
#!/usr/bin/env python
import paho.mqtt.client as paho
client = paho.Client(paho.CallbackAPIVersion.VERSION2)
client.username_pw_set("user", "pw")
client.connect("192.168.1.99", 1883, 60)

payload="01000000000000000000000000000000000000003d0000007501000077020000fd000000ae04000090040000940000001200000001000000000000000000000000000000000000003d0000007501000077020000fd000000ae040000900400009400000012000000"

client.publish("test/test1", payload)
client.disconnect()
```

**Reply by philippoo66** [reply, marked as answer]
hi @Rustimation Christoph!

The problem is not MQTT but the Optolink 'Server'. I realized with the switching times (operation schedule) that 55 bytes is the max it serves. Due to this I cut down reading the switching times to 7 (days) x 8 bytes in Viessdata. It does not harm the communication much, you usually do not read values like that every second ;-)

**Reply by Rustimation** [reply]
Ahh. Thanks for putting that straight. Since this datapoint is part of settings_ini.py, it's bound to be read every 30 seconds or any other interval. I don't know whether this is going to stress any of the SW components. If so, an alternative time schedule for seldomly read DPs would help. 
I'm slowly getting the hang of the Optolink Splitter and interpreting datapoints for my Node-Red Dashboard.
![2024-10-16 10_13_11-Centralina – Mozilla Firefox](https://github.com/user-attachments/assets/28271275-29a2-40e7-876d-7d1d663a5f8d)

Kind Rgds
Chris

**Reply by philippoo66** [reply]
looks fine!

usually, at least without Vitoconnect, the script idles 99% of the time. If really necessary, it should mean no big deal to implement a second 'instance' of the polling list / timer / mechanism. But there is some 'breath time' implemented for Optolink anyway, so the 'server' should be happy. And since you do not read dozens of datapoints in 1-second intervall, everything is cool.

when you run the skript from command line, I think you see the traffic (at least the responses) and you will realize the idling between the cycles. _If_ there is no realizable idling anymore (without Vitoconnect) then it might get time to think about a second polling instance for seldomly read datapoints.

With Vitoconnect this always bubbles all the time. Then an indication for 'too much trafic' would be that a polling cycle gets skipped (item read e.g. in 60 sec interval instead of set 30 secs).
